src/Pages/PlayQuiz/Question/Question.tsx

```diff
@@ -0,0 +1,242 @@
+            }
+        }else{
+            setChecked([id])
+        }
+    }
+
+    const nexButtonClicked=(event:SyntheticEvent)=>{
+        let score=0
+        if(checked.length>0){
+            if(multipleCorrect){
+                let correctOptions=0;
+                let selectedCorrectOptions=0
+                options.forEach(({id,isCorrect})=>{
+                    isCorrect && correctOptions++
+                    checked.includes(id!) && isCorrect && selectedCorrectOptions++
+                })
+                if(correctOptions===selectedCorrectOptions){
+                    if(showHint){
+                        score=points/2
+                    }else{
+                        score=points;
+                    }
+                    dispatch({
+                        type:"STORE_RESPONSE",
+                        payload:{
+                            content:question,
+                            response:true,
+                            index:currentIndex
+                        }
+                    })
+                }else {
+                    if(negativePoints&&negativePoints!==0){                        
+                        score=-negativePoints;
+                    }
+                    dispatch({
+                        type:"STORE_RESPONSE",
+                        payload:{
+                            content:question,
+                            response:false,
+                            index:currentIndex
+                        }
+                    })
+                }
+            }else{
+                let correctFlag=false
+                options.forEach(({id,isCorrect})=>{
+                    if(id===checked[0] && isCorrect){
+                        if(showHint){
+                            score=points/2
+                        }else{
+                            score=points;
+                        }
+                        dispatch({
+                            type:"STORE_RESPONSE",
+                            payload:{
+                                content:question,
+                                response:true,
+                                index:currentIndex
+                            }
+                        })
+                        correctFlag=true;
+                    }
+                })
+                if(!correctFlag){
+                    if(negativePoints&&negativePoints!==0){
+                        score=-negativePoints;
+                    }
+                    dispatch({
+                        type:"STORE_RESPONSE",
+                        payload:{
+                            content:question,
+                            response:false,
+                            index:currentIndex
+                        }
+                    })
+                }
+            }
+        }
+        if(((event.target as HTMLElement).innerHTML.includes("Submit"))){
```

Comment:
why is vanillaJS being used here? Any specific reason? Can't the button be clicked and onClick this check be done?

Reply:
It is done this way cause the submit function should be called after the next button function is done and code is updated. This way saves us from passing an extra argument